source/common/router/config_impl.cc

```diff
@@ -664,15 +666,23 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_host = processRequestHost(headers, final_scheme, final_port);
   }
 
+  current_path = headers.getPathValue();
   if (!path_redirect_.empty()) {
-    final_path = path_redirect_.c_str();
+    size_t path_end = current_path.find("?");
+    if (path_end == absl::string_view::npos || strip_query_) {
+      final_path = path_redirect_.c_str();
+    } else {
+      final_path_value = path_redirect_;
+      final_path_value.append(current_path.data() + path_end, current_path.length() - path_end);
```

Comment:
I expect there are existing configurations in the wild where the `redirect_path_` contains a query string but `strip_query_` is not set (because it was happening implicitly). I think the result would be `/redirectpath?redirectparams=1?originalparams=1`.

I think to handle that case, we should check for "redirect path with query string and strip query = false" at config time, emit a warning, and disable strip query (e.g. the redirect path query string takes precedence).

Reply:
that's reasonable. since I am new to this probject, I want to know if there is a best practice to emit a warning?

Reply:
is it sufficient that I simply `std::cerr << "WARNING: blabla" << std::endl`?

Reply:
fixed. PTAL!